homework/index.js

```diff
@@ -29,26 +33,149 @@
+      href: repo.html_url,
+      target: `_blank`,
+      text: repo.name,
+    });
+
+    const descriptionRow = buildTableRow(`Description`, table, ``);
+    const description = repo.description
+      ? { text: repo.description }
+      : { text: 'No Description', class: 'alert-no-description' };
+    createAndAppend(`Description`, descriptionRow.lastChild, description);
+
+    buildTableRow(`Forks`, table, repo.forks);
+
+    const date = new Date(repo.updated_at);
+    buildTableRow(`Updated`, table, date.toLocaleString());
+  }
+
+  function selectRepo(repos, select, root) {
+    const mainSection = createAndAppend('main', root, {
+      class: 'main-container',
+    });
+
+    select.addEventListener('change', () => {
+      mainSection.innerHTML = '';
+
+      const repo = repos[select.value];
+      renderRepoDetails(repo, mainSection);
+
+      const contributorsSection = createAndAppend('section', mainSection, {
+        class: 'contributors-container',
+      });
+      createAndAppend('h3', contributorsSection, { text: 'Contributions' });
+      const contributorsList = createAndAppend('ul', contributorsSection);
+
+      const CONTRIBUTORS_URL = `${HYF_BASE_URL}/repos/HackYourFuture/${repo.name}/contributors`;
+      fetchJSON(CONTRIBUTORS_URL)
+        .then(contributors =>
+          contributors.forEach(contributor =>
+            renderContributorsDetails(contributor, contributorsList),
+          ),
+        )
+
+        .catch(err =>
+          createAndAppend('div', root, {
+            text: err.message,
+            class: 'alert-error',
+          }),
+        );
+    });
   }
 
   function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
+    const root = document.getElementById('root');
+
+    const header = createAndAppend(`header`, root);
+    createAndAppend(`h1`, header, { text: `HYF Repositories` });
+    const select = createAndAppend(`select`, header);
+    createAndAppend('option', select, {
+      text: 'Select a repository',
+      disabled: 'disabled',
+      selected: 'selected',
+    });
+
+    fetchJSON(url)
+      .then(repos => {
+        repos
+          .sort((a, b) => a.name.localeCompare(b.name))
+          .forEach((repo, index) => {
+            const option = createAndAppend('option', select, { value: index });
+            option.innerHTML = repo.name;
+          });
+
+        selectRepo(repos, select, root);
+      })
+
+      .catch(err =>
         createAndAppend('div', root, {
           text: err.message,
           class: 'alert-error',
-        });
-        return;
-      }
-      const ul = createAndAppend('ul', root);
-      repos.forEach(repo => renderRepoDetails(repo, ul));
-    });
+        }),
+      );
   }
 
-  const HYF_REPOS_URL =
-    'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
-  window.onload = () => main(HYF_REPOS_URL);
+  //   const mainSection = createAndAppend(`main`, root, {
```

Comment:
Dead code like this should never be left in a PR. Nobody is interested in what you tried and abandoned. And it certainly should not makes its way into production code, if this PR were to be merged into the master branch.

Reply:
Oh, sorry. I just forgot about this piece.